src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx

```diff
@@ -168,11 +162,12 @@ export const IndexPatternTable = ({ canSave, history }: Props) => {
         setSources(dataSources.filter(removeAliases));
         setIsLoadingSources(false);
       });
-      getIndices({ http, pattern: '*:*', searchClient }).then((dataSources) =>
+      getIndices({ http, pattern: '*:*', searchClient, skipCCS: true }).then((dataSources) =>
         setRemoteClustersExist(!!dataSources.filter(removeAliases).length)
       );
     }
-  }, [http, creationOptions, searchClient, dataSourceEnabled]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
this change and skipCCS are the two changes that I dint have a lot of context on. @kristenTian What are your thoughts on this change?

Reply:
The additional context was really helpful @Ahton89. So i tried your changes and it looks like `indexPatterns.length` is not the reason for the repeated calls to the cluster, but rather your change here. The same error does not occur when multiple data sources plugin is enabled. @kristenTian @zengyan-amazon Why was this effect skipped when the datasources plugin is enabled?

Reply:
It is due to `getIndices` should not happen in here with Multiple Data Sources, but rather after user have selected a data source, OSD should then try to load the indices.